disnake/flags.py

```diff
@@ -137,6 +137,54 @@ def __eq__(self, other: Any) -> bool:
     def __ne__(self, other: Any) -> bool:
         return not self.__eq__(other)
 
+    def __and__(self, other: Self) -> Self:
+        return self._from_value(self.value & other.value)
```

Comment:
Do we want to check `isinstance` here as well?

Reply:
https://github.com/DisnakeDev/disnake/pull/605/commits/ce1421bc627c2bc5ae6d5856d4bcf5e187d99c30